Паук

KloopRE opened the pull request.

KloopRE committed:
version 1

KloopRE committed:
Доп паук

hBuzzy commented on widget.cpp:
```diff
@@ -3,14 +3,21 @@
 #include <QPainter>
 #include <QSlider>
 #include <QVBoxLayout>
+#include "widget.h"
```
Подключен два раза. Главный заголовочный отделяется от остальных пустой строкой и идет первым в подключении.

hBuzzy commented on widget.h:
```diff
@@ -13,16 +13,26 @@ class Widget : public QWidget {
 private slots:
     void moveSpider();
     void speedSliderChanged(int value);
     void spiderPositionChanged(const QPoint &position);
```
Слоты - это почти то же, что и методы. Методы всегда что-то делают -> именуем их, начиная с глагола. У вас методы:
```C++
void speedSliderChanged(int value);
void spiderPositionChanged(const QPoint &position);
```

Названы в стиле сигналов. Они не сообщают кому-то об изменении, они это новое значение обрабатывают.

hBuzzy commented on widget.h:
```diff
@@ -13,16 +13,26 @@ class Widget : public QWidget {
+    Spider *spider_;
+    int speed_;
+    QTimer *timer_;
+    bool drawing_;
```
Посмотрите замечание относительно bool переменной в задании Паутина. Тут оно аналогично.

hBuzzy commented on widget.h:
```diff
@@ -13,16 +13,26 @@ class Widget : public QWidget {
 
 protected:
```
Свои заголовочные и заголовочные Qt разделяем друг от друга пустой строкой

```с++
#include <QWidget>
#include "spider.h"
```

hBuzzy commented on spider.h:
```diff
@@ -25,18 +25,18 @@ class Spider : public QObject {
 
 public slots:
-    void moveSpider();
+    void moveSpider(const QPoint &target); // Изменим сигнатуру метода
```
``` C++
void setStartPosition(const QPoint &position);
void stopMoving();
```
Если не ошибаюсь, нигде не используются в коде. Зачем они тогда?

hBuzzy reviewed (changes requested):
Исправить замечания

KloopRE committed:
Исправление замечаний

Подключен два раза. Главный заголовочный отделяется от остальных пустой строкой и идет первым в подключении.

Слоты - это почти то же, что и методы. Методы всегда что-то делают -> именуем их, начиная с глагола.

 замечание относительно bool переменной в задании Паутина.

Свои заголовочные и заголовочные Qt разделяем друг от друга пустой строкой

Widget::moveSpider():

Этот метод вызывается по таймеру и перемещает паука в зависимости от текущего положения курсора мыши (center_).
Он также устанавливает флаг isdrawing_ в true, что, видимо, означает, что паук находится в режиме рисования.
Spider::moveSpider(const QPoint &target):

Этот метод перемещает паука к заданной точке target.
Он используется для перемещения паука в ответ на события мыши или другие события, когда позиция паука должна измениться.
Иными словами, Widget::moveSpider() ответственен за обновление положения паука в соответствии с положением курсора мыши, а Spider::moveSpider(const QPoint &target) позволяет перемещать паука в явно заданные точки.

hBuzzy commented on widget.h:
```diff
@@ -29,7 +29,7 @@ private slots:
     int speed_;
     QTimer *timer_;
-    bool drawing_;
+    bool isdrawing_;
```
isdrawing_ -> isDrawing_

hBuzzy reviewed (commented):
Принято.

KloopRE committed:
Доп

hBuzzy commented on widget.cpp:
```diff
@@ -105,3 +109,32 @@ void Widget::mouseReleaseEvent(QMouseEvent *event) {
 }
+
+void Widget::keyPressEvent(QKeyEvent *event) {
+    if (isdrawing_) {
```
isDrawing_

hBuzzy reviewed (commented):
Принято.
